### ISSUE TITLE
auth-mongo: DefaultHashStrategy creates lowercase hashes in 4.2, before they were uppercase

### ISSUE THREAD

**Issue by calohmn** [eclipse-vertx/vertx-auth#534]
### Version

vertx-auth-mongo 4.2.4

### Context

We noticed that after having updated from vert.x 4.1 to 4.2, our MongoDB client couldn't login anymore using the credentials that worked before. We use `io.vertx.ext.auth.mongo.impl.MongoAuthenticationImpl` with a `DefaultHashStrategy` as legacy hash strategy.

Looking at the root cause, this seems to be caused by the way hashes are created by `io.vertx.ext.auth.mongo.impl.DefaultHashStrategy#computeHash(java.lang.String, io.vertx.ext.auth.User)`.

In vert.x <= 4.1, the created hashes contained uppercase letters.

In vert.x 4.2, hashes contain lowercase hashes.

### Steps to reproduce

```
    DefaultHashStrategy strategy = new DefaultHashStrategy();
    strategy.setSaltStyle(HashSaltStyle.EXTERNAL);
    strategy.setExternalSalt("salt");
    System.out.println(strategy.computeHash("Secret", null));
```
**Output using vertx-auth-mongo 4.1:**
```
A110329997FE95234BB2E91A547EFE2FB5CC3018C159E32FFDC4230FA2189B7C062D4B02A3E2445389E80C78BA7627766F6B2369F1137EB52C4A0F7B4460F3D2
```
=> uppercase

**Output using vertx-auth-mongo 4.2.4:**
```
a110329997fe95234bb2e91a547efe2fb5cc3018c159e32ffdc4230fa2189b7c062d4b02a3e2445389e80c78ba7627766f6b2369f1137eb52c4a0f7b4460f3d2
```
=> lowercase

### Extra

This change seems to have been introduced by #509.

**Comment by pmlopes** [comment]
@calohmn I believe the bug is that I forgot to uppercase the SHA512 case.

https://github.com/vert-x3/vertx-auth/pull/537/files

Can you confirm this? If this is the fix, I'll push asap to master to make the 4.2.5 release!

**Comment by calohmn** [comment]
@pmlopes yes, this seems to fix it.
